rest/route/keys_routes.go

```diff
@@ -140,6 +140,13 @@ func validateKeyValue(keyValue string) error {
 func (h *keysPostHandler) Execute(ctx context.Context, sc data.Connector) (ResponseData, error) {
 	u := MustHaveUser(ctx)
 
+	if _, err := u.GetPublicKey(h.keyName); err == nil {
+		return ResponseData{}, &rest.APIError{
+			StatusCode: http.StatusBadRequest,
+			Message:    "duplicate key name",
```

Comment:
This is kind of a nit. I would use a different error here, as this is basically the same as what happens with unique indexes. I'd say "a key with this name is already defined for this user.